src/Features/Core/Portable/DocumentationComments/AbstractDocumentationCommentFormattingService.cs

```diff
@@ -102,10 +103,12 @@ public void AppendSingleSpace()
             public void AppendString(string s)
             {
                 EmitPendingChars();
-
-                Builder.Add(new TaggedText(TextTags.Text, s, Style, NavigationTarget.target, NavigationTarget.hint));
+                Builder.Add(new TaggedText(TextTags.Text, NormalizeLineEndings(s), Style, NavigationTarget.target, NavigationTarget.hint));
 
                 _anyNonWhitespaceSinceLastPara = true;
+
+                static string NormalizeLineEndings(string input) => Regex.Replace(input, "(?<!\r)\n", "\r\n");
```

Comment:
I'd certainly prefer a better solution here. The issue here is that `XText.Value` produces a string with Unix-style line endings (`\n`), causing the end result to have mixed line-endings 😕

Reply:
this warrants a comment explaning what the regex means and waht this is here for.

Reply:
I normally use `input.Replace("\r\n", "\n").Replace("\n", Environment.NewLine)`, but XML is required to normalize line endings to `\n` so I'm guessing you can just use `input.Replace("\n", "\r\n")` and be done.

Reply:
@sharwell I'm a little bit worried if AppendString can be called with a string already containing `\r\n`, then doing `Replace("\n", "\r\n")` will duplicate `\r`.

Is it's guaranteed that `XText.Value` will **always** be normalized to `\n`?